### DISCUSSION TITLE
Versions compatible with each other?

### DISCUSSION THREAD

**Discussion by AGI-chandler** [Xpra-org/xpra#3717] [Q&A]
I have a range of OS and kernels that I have to work with, some as old as RHEL 6, Linux 2.6.  For that particular environment, the latest xpra available in your repos is v1.0.14-r24088.  Unfortunately that's not compatible with the v4.4.3 that I have installed on my workstation, soon to be 5.0 it looks like.  What is it compatible with, then?  Only another v1.x xpra or what?

Is there a recommended way to have two (or more) different versions of xpra installed on my workstation? chroot maybe?

Thanks

**Comment by totaam** [comment]
> the latest xpra available in your repos is `v1.0.14-r24088`
> (..)
> Unfortunately that's not compatible with the v4.4.3 that I have installed on my workstation

What was the error?
It may be possible to make it work.
But don't expect miracles, RHEL 6.x is just too old.
> (..)
> What is it compatible with, then?  Only another v1.x xpra or what?

IIRC, it was compatible all the way to v4.x something.
v3.x should be compatible with it for sure.

> Is there a recommended way to have two (or more) different versions of xpra installed on my workstation? chroot maybe?

Recommended, no.
Possible, definitely.
There are many different ways of doing this, it all really depends on the tools at your disposal.

For reference:
https://github.com/Xpra-org/xpra/wiki/Versions/_history
https://github.com/Xpra-org/xpra/wiki/Versions/e486cb66261b90cd6748852ec72d5a9f18bb27aa

**Reply by AGI-chandler** [reply]
> What was the error? It may be possible to make it work. But don't expect miracles, RHEL 6.x is just too old.
> 
It just says: `incompatible remote version '1.0.14': remote version [1, 0] is too old, sorry`

> Recommended, no. Possible, definitely. There are many different ways of doing this, it all really depends on the tools at your disposal.
> 
My workstation is mine so I can install/run anything...

**Reply by AGI-chandler** [reply]
Well I just tried downgrading until I found one that works, **4.3.4-r0-1** is compatible with 1.0.14-1.r24088.el6_10 and 4.4.3-10.el8 so that should be sufficient for now...

**Reply by totaam** [reply]
Ah, right.
Now I remember changing this in 5dbad5da25ac2191815397d13c2251a5ff2da2fa
The 1.0.x branch has been abandoned for years now.
